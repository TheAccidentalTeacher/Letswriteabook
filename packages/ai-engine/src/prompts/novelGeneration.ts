import { advancedHumanWritingRefinements } from './advancedHumanWritingRefinements';
import { genreInstructions } from './genreInstructions';
import { humanWritingEnhancements } from './humanWritingEnhancements';
import { universalHumanWritingFramework } from './universalHumanWritingFramework';

export interface PremiseAnalysisPromptInput {
  readonly premise: string;
  readonly genre: string;
  readonly subgenre: string;
  readonly targetWordCount: number;
  readonly targetChapters: number;
  readonly humanLikeWriting: boolean;
}

export interface OutlinePromptInput {
  readonly title: string;
  readonly premise: string;
  readonly genre: string;
  readonly subgenre: string;
  readonly targetWordCount: number;
  readonly targetChapters: number;
  readonly humanLikeWriting: boolean;
}

export interface ChapterOutlineDetails {
  readonly chapterNumber: number;
  readonly title: string;
  readonly summary: string;
  readonly keyEvents: readonly string[];
  readonly wordTarget: number;
  readonly humanLikeElements?: Record<string, unknown>;
}

export interface ExistingChapterSummary {
  readonly chapterNumber: number;
  readonly title: string;
  readonly wordCount?: number | null;
}

export interface AnalysisSummary {
  readonly themes?: readonly string[];
  readonly characters?: readonly unknown[];
  readonly humanLikeElements?: Record<string, unknown>;
  readonly [key: string]: unknown;
}

export interface ChapterPromptInput {
  readonly title: string;
  readonly premise: string;
  readonly genre: string;
  readonly subgenre: string;
  readonly targetChapters: number;
  readonly humanLikeWriting: boolean;
  readonly analysis?: AnalysisSummary | null;
  readonly chaptersSoFar: ReadonlyArray<ExistingChapterSummary>;
  readonly chapterOutline: ChapterOutlineDetails;
}

function formatLabel(value: string): string {
  return value.replace(/_/g, ' ');
}

function resolveGenreInstruction(genre: string, subgenre: string): string {
  const byExactGenre = (genreInstructions as Record<string, Record<string, string> | undefined>)[genre]?.[subgenre];
  if (byExactGenre) {
    return byExactGenre;
  }

  const normalizedGenre = genre.toUpperCase();
  const normalizedSubgenre = subgenre.toUpperCase();
  const byNormalizedGenre = (genreInstructions as Record<string, Record<string, string> | undefined>)[normalizedGenre]?.[normalizedSubgenre];
  if (byNormalizedGenre) {
    return byNormalizedGenre;
  }

  throw new Error(`Unsupported genre combination: ${genre}/${subgenre}`);
}

function formatThemes(analysis?: AnalysisSummary | null): string {
  if (!analysis || !Array.isArray(analysis.themes) || analysis.themes.length === 0) {
    return 'N/A';
  }

  return analysis.themes.join(', ');
}

function formatCharacters(analysis?: AnalysisSummary | null): string {
  if (!analysis || !Array.isArray(analysis.characters) || analysis.characters.length === 0) {
    return 'N/A';
  }

  return analysis.characters.map((character) => String(character)).join(', ');
}

export function buildPremiseAnalysisPrompt(input: PremiseAnalysisPromptInput): string {
  const genreInstruction = resolveGenreInstruction(input.genre, input.subgenre);

  return `
Analyze this novel premise and provide structural recommendations:

PREMISE: "${input.premise}"

GENRE: ${formatLabel(input.genre)}
SUBGENRE: ${formatLabel(input.subgenre)}
TARGET WORD COUNT: ${input.targetWordCount}
TARGET CHAPTERS: ${input.targetChapters}

GENRE GUIDELINES:
${genreInstruction}

${input.humanLikeWriting ? humanWritingEnhancements.prompts.analysis.humanLikeAdditions : ''}

${input.humanLikeWriting ? universalHumanWritingFramework.promptEnhancements.analysis : ''}

${input.humanLikeWriting ? advancedHumanWritingRefinements.level5Prompts.analysis : ''}

Please provide a comprehensive analysis that prioritizes ${input.humanLikeWriting ? 'authentic, human-like storytelling' : 'engaging storytelling'}:

ANALYSIS REQUIREMENTS:
1. Theme analysis - themes that allow for moral complexity
2. Character archetypes - with internal contradictions and growth potential
3. Plot structure - that accommodates meaningful failures and setbacks
4. Key story beats - including moments of genuine uncertainty
5. Potential subplots - that may remain partially unresolved
6. Tone and style guidance - that varies subtly throughout
${input.humanLikeWriting ? `7. HUMAN-LIKE ELEMENTS:
   - Internal character conflicts that create lasting tension
   - Opportunities for protagonist to be genuinely wrong
   - Morally ambiguous situations requiring difficult choices
   - Cultural/world elements that add lived-in authenticity
   - Distinctive character voice planning (speech patterns, vocabulary)` : `7. ENGAGING ELEMENTS:
   - Clear character motivations and goals
   - Compelling conflicts and obstacles
   - Satisfying story progression
   - Genre-appropriate atmosphere and tone`}

Respond in JSON format:
{
  "themes": ["theme1", "theme2"],
  "characters": [{"type": "character_type", "conflicts": "internal_struggles", "speechPattern": "distinctive_traits"}],
  "plotStructure": "structure_with_flexibility_for_messiness",
  "keyBeats": ["beat1", "beat2"],
  "subplots": [{"main": "subplot", "resolution": "complete|partial|unresolved"}],
  "tone": "description_that_allows_variation",
  "styleNotes": "guidance_for_character_specific_prose"${input.humanLikeWriting ? `,
  "humanLikeElements": {
    "characterConflicts": ["lasting_disagreement1", "lasting_disagreement2"],
    "moralDilemmas": ["situation1", "situation2"],
    "culturalElements": ["in_world_reference1", "in_world_reference2"],
    "unresolvedElements": ["mystery1", "relationship_tension1"]
  }` : ''}
}`;
}

export function buildOutlinePrompt(input: OutlinePromptInput): string {
  const averageWordsPerChapter = Math.round(input.targetWordCount / input.targetChapters);

  return `
Create a ${input.targetChapters}-chapter outline for "${input.title}" (${formatLabel(input.genre)} - ${formatLabel(input.subgenre)}).

SYNOPSIS: ${input.premise}
WORD COUNT: ${input.targetWordCount} total (~${averageWordsPerChapter} per chapter)

${input.humanLikeWriting ? humanWritingEnhancements.prompts.outline.humanLikeAdditions : ''}

${input.humanLikeWriting ? universalHumanWritingFramework.promptEnhancements.outline : ''}

${input.humanLikeWriting ? advancedHumanWritingRefinements.level5Prompts.outline : ''}

${input.humanLikeWriting ? `HUMAN-LIKE OUTLINE REQUIREMENTS:
- Create significant variation in chapter lengths (some 800 words, others 3000+ words)
- Mix chapter types: action, dialogue-heavy, introspective, world-building focused
- Plan at least 2 meaningful character failures that don't immediately resolve
- Include 1-2 chapters with unresolved endings that complicate rather than clarify
- Design at least one major plot twist that genuinely surprises (not just reveals)
- Ensure internal character conflicts span multiple chapters without easy resolution
- Plan chapters that show the same events from different character perspectives
- Include at least one chapter told through non-traditional format (logs, messages, flashbacks)` : `OUTLINE REQUIREMENTS:
- Create engaging chapter progression with clear story beats
- Build tension and character development throughout
- Include compelling conflicts and resolutions
- Maintain genre conventions and reader expectations`}

Create exactly ${input.targetChapters} chapters with detailed descriptions that embrace ${input.humanLikeWriting ? 'narrative complexity' : 'engaging storytelling'}:

JSON format:
{
  "outline": [
    {
      "chapterNumber": 1,
      "title": "Chapter Title",
      "summary": "Key events and plot progression",
      "keyEvents": ["event1", "event2", "event3"],
      "characterFocus": ["char1", "char2"],
      "plotAdvancement": "How this chapter advances or complicates the main plot",
      "wordTarget": ${averageWordsPerChapter},
      "genreElements": ["genre-specific element1", "genre-specific element2"]${input.humanLikeWriting ? `,
      "humanLikeElements": {
        "structureType": "traditional|dialogue-heavy|introspective|action|logs|flashback",
        "characterConflict": "internal_or_interpersonal_tension_introduced_or_developed",
        "moralComplexity": "ethical_dilemma_or_ambiguous_situation",
        "unresolvedElement": "something_left_hanging_or_complicated",
        "surpriseElement": "unexpected_development_or_character_choice",
        "mundaneDetail": "lived_in_world_element_that_adds_authenticity"
      }` : ''}
    }
  ]
}`;
}

export function buildChapterPrompt(input: ChapterPromptInput): string {
  const { chapterOutline } = input;
  const keyEvents = Array.isArray(chapterOutline.keyEvents) && chapterOutline.keyEvents.length > 0
    ? chapterOutline.keyEvents.join(', ')
    : 'Chapter events to be determined';
  const previousChapters = input.chaptersSoFar.length > 0
    ? input.chaptersSoFar
        .slice(-3)
        .map((chapter) => `Ch${chapter.chapterNumber}: ${chapter.title} (${String(chapter.wordCount)}w)`)
        .join('; ')
    : 'This is the first chapter';
  const analysisContext = input.targetChapters > 20
    ? `Key themes: ${formatThemes(input.analysis)}
Main characters: ${formatCharacters(input.analysis)}${input.humanLikeWriting ? `
Human-like story elements: ${JSON.stringify(input.analysis?.humanLikeElements ?? {}, null, 1)}` : ''}`
    : JSON.stringify(input.analysis ?? {}, null, 1);
  const genreInstruction = resolveGenreInstruction(input.genre, input.subgenre);

  return `
Write Chapter ${chapterOutline.chapterNumber} of the novel "${input.title}".

CHAPTER OUTLINE:
Title: ${chapterOutline.title}
Summary: ${chapterOutline.summary}
Key Events: ${keyEvents}
Target Word Count: ${chapterOutline.wordTarget}${input.humanLikeWriting ? `
Human-Like Elements: ${JSON.stringify(chapterOutline.humanLikeElements ?? {}, null, 1)}` : ''}

NOVEL CONTEXT:
Premise: "${input.premise}"
Genre: ${formatLabel(input.genre)} - ${formatLabel(input.subgenre)}
Previous chapters: ${previousChapters}
Story progress: Chapter ${chapterOutline.chapterNumber} of ${input.targetChapters} total

GENRE GUIDELINES:
${genreInstruction}

ANALYSIS CONTEXT:
${analysisContext}

${input.humanLikeWriting ? humanWritingEnhancements.prompts.chapter.humanLikeAdditions : ''}

${input.humanLikeWriting ? universalHumanWritingFramework.promptEnhancements.chapter : ''}

${input.humanLikeWriting ? advancedHumanWritingRefinements.level5Prompts.chapter : ''}

${input.humanLikeWriting ? `LEVEL 5 ADVANCED HUMAN AUTHENTICITY IMPLEMENTATION:

FACTION BREAKING POINT - INTERNAL CONFLICTS WITH CONSEQUENCES:
- Show ally factions sabotaging protagonist's plans from genuine good intentions (no clear villains)
- Force leader into choice between authoritarian control and collaborative risk with lasting consequences
- Make internal division as dangerous as external threats - competing survival needs drive conflict
- Create "hard choice" scenarios where every option creates lasting community division

POWER COST ENFORCEMENT - MEANINGFUL LIMITATIONS:
- After significant ability use, show character suffering real biological/psychological costs (memory loss, exhaustion, vulnerability)
- Include empathetic feedback where protector characters physically/emotionally feel pain when their charges suffer
- Create recovery periods where powerful characters are dependent on others and cannot use abilities
- Never let powers solve problems without exacting meaningful prices that affect story progression

STRUCTURAL SUBVERSION - BREAK ESTABLISHED PATTERNS:
- Avoid repetitive meeting/discussion locations (break cantina meeting pattern if established)
- Include critical decision made by isolated character with no time for group consultation
- Show how small character weakness or overlooked detail catastrophically derails simple plans
- Create genuine surprise that feels inevitable once revealed

ACTIVE PROSE CONSTRAINTS - FORBIDDEN PHRASE ENFORCEMENT:
- ABSOLUTELY FORBIDDEN: "silver-flecked eyes," "copper-plated exosuit," "vibrant hair," "living suit," "multi-tool gauntlet," "weathered face," "steely determination," "nanite cloak," "luminescent ink"
- Describe characters/equipment through action and environmental effect: "weight settled across shoulders" not "heavy exosuit"
- Use fresh, specific details that haven't appeared in previous chapters
- Show character traits through behavior and dialogue, not repetitive physical descriptions

ADVANCED AUTHENTICITY TECHNIQUES:
- Let established character strengths become weaknesses under different pressures
- Show internal faction politics affecting every decision, not just major plot points
- Include moments where characters surprise themselves with choices under pressure
- Create dialogue subtext that contradicts spoken words, revealing deeper faction conflicts
- Make consequences of past chapters continue to affect current events and relationships

Write approximately ${chapterOutline.wordTarget} words that push established complexity to breaking points and challenge reader expectations while maintaining narrative authenticity.` : `Write approximately ${chapterOutline.wordTarget} words of engaging prose that maintains genre conventions and advances the story effectively.`}

Write only the chapter content, no metadata or formatting.`;
}
