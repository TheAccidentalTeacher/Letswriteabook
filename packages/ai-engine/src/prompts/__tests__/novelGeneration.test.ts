import { buildChapterPrompt, buildOutlinePrompt, buildPremiseAnalysisPrompt } from '../novelGeneration';

describe('novel generation prompts', () => {
  it('builds premise analysis prompt with human-like enhancements', () => {
    const prompt = buildPremiseAnalysisPrompt({
      premise: 'An exiled starship captain must broker peace between warring factions while confronting their own legacy.',
      genre: 'SCIENCE_FICTION',
      subgenre: 'SPACE_OPERA',
      targetWordCount: 90000,
      targetChapters: 24,
      humanLikeWriting: true,
    });

    expect(prompt).toMatchSnapshot();
  });

  it('builds outline prompt with human-like storytelling requirements', () => {
    const prompt = buildOutlinePrompt({
      title: 'Shattered Orbit',
      premise: 'A scattered rebellion races to unify before an ancient AI awakens a doomsday fleet.',
      genre: 'SCIENCE_FICTION',
      subgenre: 'SPACE_OPERA',
      targetWordCount: 90000,
      targetChapters: 24,
      humanLikeWriting: true,
    });

    expect(prompt).toMatchSnapshot();
  });

  it('builds chapter prompt with full human authenticity guidance', () => {
    const prompt = buildChapterPrompt({
      title: 'Shattered Orbit',
      premise: 'A scattered rebellion races to unify before an ancient AI awakens a doomsday fleet.',
      genre: 'SCIENCE_FICTION',
      subgenre: 'SPACE_OPERA',
      targetChapters: 24,
      humanLikeWriting: true,
      analysis: {
        themes: ['Legacy', 'Collective Responsibility'],
        characters: ['Captain Mira Solari', 'AI Custodian Arcturus'],
        humanLikeElements: {
          characterConflicts: ['Leadership fractures', 'Ethical compromises'],
        },
      },
      chaptersSoFar: [
        { chapterNumber: 1, title: 'Prologue: The Silent Call', wordCount: 2500 },
        { chapterNumber: 2, title: 'Fragments of Trust', wordCount: 3100 },
        { chapterNumber: 3, title: 'Echoes in the Dust', wordCount: 2800 },
      ],
      chapterOutline: {
        chapterNumber: 4,
        title: 'Negotiations in Low Orbit',
        summary: 'Mira attempts to reconcile two rival cells while an AI emissary tests their resolve.',
        keyEvents: [
          'Arrival aboard the scarred carrier',
          'Heated debate collapsing into factional accusations',
          'Unexpected AI transmission forces new terms',
        ],
        wordTarget: 3400,
        humanLikeElements: {
          structureType: 'dialogue-heavy',
          characterConflict: 'Mira vs. former co-captain over mission priorities',
          moralComplexity: 'Choosing between sacrificing allies or civilians',
          unresolvedElement: 'AI ultimatum left hanging',
          surpriseElement: 'Revealed spy in Miraâ€™s escort crew',
          mundaneDetail: 'Ship vents rattling from neglected maintenance',
        },
      },
    });

    expect(prompt).toMatchSnapshot();
  });
});
