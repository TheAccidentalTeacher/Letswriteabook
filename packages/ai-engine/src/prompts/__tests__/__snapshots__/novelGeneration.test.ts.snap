// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`novel generation prompts builds chapter prompt with full human authenticity guidance 1`] = `
"
Write Chapter 4 of the novel "Shattered Orbit".

CHAPTER OUTLINE:
Title: Negotiations in Low Orbit
Summary: Mira attempts to reconcile two rival cells while an AI emissary tests their resolve.
Key Events: Arrival aboard the scarred carrier, Heated debate collapsing into factional accusations, Unexpected AI transmission forces new terms
Target Word Count: 3400
Human-Like Elements: {
 "structureType": "dialogue-heavy",
 "characterConflict": "Mira vs. former co-captain over mission priorities",
 "moralComplexity": "Choosing between sacrificing allies or civilians",
 "unresolvedElement": "AI ultimatum left hanging",
 "surpriseElement": "Revealed spy in Miraâ€™s escort crew",
 "mundaneDetail": "Ship vents rattling from neglected maintenance"
}

NOVEL CONTEXT:
Premise: "A scattered rebellion races to unify before an ancient AI awakens a doomsday fleet."
Genre: SCIENCE FICTION - SPACE OPERA
Previous chapters: Ch1: Prologue: The Silent Call (2500w); Ch2: Fragments of Trust (3100w); Ch3: Echoes in the Dust (2800w)
Story progress: Chapter 4 of 24 total

GENRE GUIDELINES:
Focus on grand-scale adventures across galaxies. Include interstellar travel, multiple alien civilizations, epic conflicts, and advanced technology. Emphasize character-driven narratives against the backdrop of a vast universe. Balance action with philosophical themes about humanity's place in the cosmos.

ANALYSIS CONTEXT:
Key themes: Legacy, Collective Responsibility
Main characters: Captain Mira Solari, AI Custodian Arcturus
Human-like story elements: {
 "characterConflicts": [
  "Leadership fractures",
  "Ethical compromises"
 ]
}


HUMAN-LIKE CHAPTER WRITING REQUIREMENTS:

CHARACTER AUTHENTICITY:
- Each character should have distinctive speech patterns, verbal tics, and vocabulary choices
- Include subtext in dialogue - characters saying one thing but meaning another
- Show characters making unexpected but believable choices that reveal contradictions
- Avoid repetitive character descriptions (same traits mentioned repeatedly)
- Include genuine interpersonal friction that doesn't resolve easily
- Show characters failing meaningfully without immediate plot advancement

NARRATIVE REALISM:
- Add mundane frustrations: equipment breaking, miscommunications, physical discomfort
- Include sensory inconsistencies - characters perceiving same events differently
- Avoid over-explaining themes or having characters speak too philosophically
- Let conflicts be messy and not resolve too neatly or conveniently
- Show genuine surprise for protagonists - they should be caught off-guard
- Avoid too many profound or scripted-sounding conversations

WORLD DEPTH:
- Reference specific in-world cultural elements (songs, sayings, shared history)
- Include organic humor that emerges naturally from situations
- Add small equipment failures or everyday complications at inconvenient times
- Show characters dealing with mundane annoyances that compound
- Vary sensory details - different smells, tastes, textures, sounds

STRUCTURAL VARIETY:
- Avoid formulaic chapter structure (setup-conflict-resolution)
- Vary prose style based on which character is the focus
- Include unexpected moments that don't serve the main plot
- End with something unresolved or complicating rather than neat closure
- Break expected narrative patterns with genuine surprises
- Mix chapter formats - some action-heavy, others reflective, some through logs/messages

ANTAGONIST COMPLEXITY:
- Make antagonists more complex with understandable motivations beyond simple greed
- Show opposition that makes valid points even when wrong overall
- Create conflicts where there's no clear "right" answer
- Develop institutional/corporate enemies with human faces and relatable reasons


UNIVERSAL FRAMEWORK FOR CHAPTER WRITING:

CHARACTER FALLIBILITY IN ACTION:
- Show characters making decisions driven by their core psychological flaws
- Include meaningful failures that have real, lasting consequences beyond this chapter
- Let character strengths become weaknesses when taken to extremes

PROSE AUTHENTICITY REQUIREMENTS:
- Avoid repetitive character descriptors (forbidden words: silver-flecked, weathered, steely)
- Describe through action and effect rather than direct description
- Include mundane annoyances that don't advance plot but add authenticity

DIALOGUE WITH LAYERS:
- Every dialogue scene must have subtext - characters wanting something they won't say
- Use body language and actions that contradict spoken words
- Show character relationships through what they DON'T say to each other

WORLD-BUILDING THROUGH LIVED-IN DETAILS:
- Show world history through worn objects and environmental details
- Include minor equipment failures and everyday complications
- Ground extraordinary elements in relatable, mundane problems

ANTAGONIST HUMANIZATION:
- If antagonist appears, show their perspective and understandable motivations
- Make opposition that raises valid points even when wrong overall
- Avoid "evil for evil's sake" - show the human reasoning behind opposition


LEVEL 5 ADVANCED FRAMEWORK FOR CHAPTER WRITING:

FACTION BREAKING POINT IMPLEMENTATION:
- If internal conflict appears, show ally faction sabotaging protagonist from genuine good intentions
- Create "no clear villain" scenarios where competing survival needs drive conflict
- Show leader forced to choose between crushing dissent and risking mission failure
- Make consequences of internal division feel as real and dangerous as external threats

POWER COST ENFORCEMENT:
- After any significant ability use, show character suffering real biological/psychological costs
- Include empathetic feedback where protector characters feel pain when their charges suffer
- Create recovery periods where powerful characters are vulnerable and dependent on others
- Never let abilities solve problems without exacting meaningful prices

STRUCTURAL SUBVERSION EXECUTION:
- Break established pattern if one has emerged (avoid repetitive meeting locations/formats)
- Include moment where critical decision must be made by isolated character with no consultation
- Show how small character weakness or overlooked detail can catastrophically derail simple plans
- Create genuine surprise that feels inevitable once revealed

ACTIVE PROSE CONSTRAINTS:
- FORBIDDEN PHRASES: "silver-flecked eyes," "copper-plated exosuit," "vibrant hair," "living suit," "multi-tool gauntlet," "weathered face," "steely determination," "nanite cloak," "luminescent ink"
- Describe characters and equipment through action, environmental effect, and sensory impact
- Use fresh, specific details that haven't appeared in previous chapters
- Show character traits through behavior and dialogue rather than physical descriptions

ADVANCED AUTHENTICITY TECHNIQUES:
- Let established character strengths become weaknesses under different pressures
- Show internal faction politics affecting every decision, not just major plot points
- Include moments where characters surprise themselves with their choices under pressure
- Create dialogue where subtext contradicts spoken words, revealing deeper conflicts

LEVEL 5 ADVANCED HUMAN AUTHENTICITY IMPLEMENTATION:

FACTION BREAKING POINT - INTERNAL CONFLICTS WITH CONSEQUENCES:
- Show ally factions sabotaging protagonist's plans from genuine good intentions (no clear villains)
- Force leader into choice between authoritarian control and collaborative risk with lasting consequences
- Make internal division as dangerous as external threats - competing survival needs drive conflict
- Create "hard choice" scenarios where every option creates lasting community division

POWER COST ENFORCEMENT - MEANINGFUL LIMITATIONS:
- After significant ability use, show character suffering real biological/psychological costs (memory loss, exhaustion, vulnerability)
- Include empathetic feedback where protector characters physically/emotionally feel pain when their charges suffer
- Create recovery periods where powerful characters are dependent on others and cannot use abilities
- Never let powers solve problems without exacting meaningful prices that affect story progression

STRUCTURAL SUBVERSION - BREAK ESTABLISHED PATTERNS:
- Avoid repetitive meeting/discussion locations (break cantina meeting pattern if established)
- Include critical decision made by isolated character with no time for group consultation
- Show how small character weakness or overlooked detail catastrophically derails simple plans
- Create genuine surprise that feels inevitable once revealed

ACTIVE PROSE CONSTRAINTS - FORBIDDEN PHRASE ENFORCEMENT:
- ABSOLUTELY FORBIDDEN: "silver-flecked eyes," "copper-plated exosuit," "vibrant hair," "living suit," "multi-tool gauntlet," "weathered face," "steely determination," "nanite cloak," "luminescent ink"
- Describe characters/equipment through action and environmental effect: "weight settled across shoulders" not "heavy exosuit"
- Use fresh, specific details that haven't appeared in previous chapters
- Show character traits through behavior and dialogue, not repetitive physical descriptions

ADVANCED AUTHENTICITY TECHNIQUES:
- Let established character strengths become weaknesses under different pressures
- Show internal faction politics affecting every decision, not just major plot points
- Include moments where characters surprise themselves with choices under pressure
- Create dialogue subtext that contradicts spoken words, revealing deeper faction conflicts
- Make consequences of past chapters continue to affect current events and relationships

Write approximately 3400 words that push established complexity to breaking points and challenge reader expectations while maintaining narrative authenticity.

Write only the chapter content, no metadata or formatting."
`;

exports[`novel generation prompts builds outline prompt with human-like storytelling requirements 1`] = `
"
Create a 24-chapter outline for "Shattered Orbit" (SCIENCE FICTION - SPACE OPERA).

SYNOPSIS: A scattered rebellion races to unify before an ancient AI awakens a doomsday fleet.
WORD COUNT: 90000 total (~3750 per chapter)


STRUCTURAL VARIETY REQUIREMENTS:
- Vary chapter lengths significantly (some short, some long)
- Mix chapter types: action, reflection, dialogue-heavy, description-heavy, logs/messages
- Plan uneven pacing with slower moments that build character
- Ensure some chapters end without clear resolution or neat closure
- Plan at least one major plot twist that genuinely surprises characters
- Include meaningful failures that create genuine setbacks without easy recovery
- Create chapters where equipment fails or miscommunications cause problems
- Plan for different characters to perceive same events differently
- Include at least one chapter focused on mundane frustrations and complications
- Avoid formulaic chapter structures (setup-conflict-resolution pattern)


UNIVERSAL FRAMEWORK FOR OUTLINE STRUCTURE:

PACING AND REVELATION MASTERY:
- Use Rule of Three: introduce key elements three times before they become critical
- Plant seeds for major events through environmental clues and character reactions
- Plan foreshadowing that feels natural, not forced or obvious

AMBIGUITY AND UNRESOLVED ELEMENTS:
- Leave secondary plot points and some character fates deliberately unresolved
- Create moral and philosophical questions without clear answers
- Plan chapter endings that pose lingering questions about character morality

COMPLEX MORAL DILEMMAS:
- Design no-win scenarios where every choice has significant cost
- Challenge genre tropes with morally complex alternatives
- Create situations where traditional 'good' choices become ambiguous


LEVEL 5 ADVANCED FRAMEWORK FOR OUTLINE:

FACTION CONFLICT ESCALATION:
- Plan specific sabotage scenarios where ally factions act against protagonist's plans
- Design "hard choice" moments where leader must choose between authoritarian control and collaborative risk
- Structure lasting consequences that affect community dynamics throughout remaining story
- Ensure internal conflicts are as dangerous as external threats

POWER LIMITATION INTEGRATION:
- Plan major power usage moments with specific, debilitating aftermath effects
- Design empathetic connection scenes where character suffers when protected area is harmed
- Structure power costs that force characters to use abilities sparingly and strategically
- Create vulnerability windows when powerful characters are recovering from ability use

NARRATIVE STRUCTURE SUBVERSION:
- Plan decentralized decision moments where isolated characters make story-changing choices
- Design "simple mission goes wrong" scenarios that become major plot developments
- Break established meeting/discussion patterns with urgent, pressure-cooker decisions
- Create unpredictable story beats that still feel inevitable in hindsight

HUMAN-LIKE OUTLINE REQUIREMENTS:
- Create significant variation in chapter lengths (some 800 words, others 3000+ words)
- Mix chapter types: action, dialogue-heavy, introspective, world-building focused
- Plan at least 2 meaningful character failures that don't immediately resolve
- Include 1-2 chapters with unresolved endings that complicate rather than clarify
- Design at least one major plot twist that genuinely surprises (not just reveals)
- Ensure internal character conflicts span multiple chapters without easy resolution
- Plan chapters that show the same events from different character perspectives
- Include at least one chapter told through non-traditional format (logs, messages, flashbacks)

Create exactly 24 chapters with detailed descriptions that embrace narrative complexity:

JSON format:
{
  "outline": [
    {
      "chapterNumber": 1,
      "title": "Chapter Title",
      "summary": "Key events and plot progression",
      "keyEvents": ["event1", "event2", "event3"],
      "characterFocus": ["char1", "char2"],
      "plotAdvancement": "How this chapter advances or complicates the main plot",
      "wordTarget": 3750,
      "genreElements": ["genre-specific element1", "genre-specific element2"],
      "humanLikeElements": {
        "structureType": "traditional|dialogue-heavy|introspective|action|logs|flashback",
        "characterConflict": "internal_or_interpersonal_tension_introduced_or_developed",
        "moralComplexity": "ethical_dilemma_or_ambiguous_situation",
        "unresolvedElement": "something_left_hanging_or_complicated",
        "surpriseElement": "unexpected_development_or_character_choice",
        "mundaneDetail": "lived_in_world_element_that_adds_authenticity"
      }
    }
  ]
}"
`;

exports[`novel generation prompts builds premise analysis prompt with human-like enhancements 1`] = `
"
Analyze this novel premise and provide structural recommendations:

PREMISE: "An exiled starship captain must broker peace between warring factions while confronting their own legacy."

GENRE: SCIENCE FICTION
SUBGENRE: SPACE OPERA
TARGET WORD COUNT: 90000
TARGET CHAPTERS: 24

GENRE GUIDELINES:
Focus on grand-scale adventures across galaxies. Include interstellar travel, multiple alien civilizations, epic conflicts, and advanced technology. Emphasize character-driven narratives against the backdrop of a vast universe. Balance action with philosophical themes about humanity's place in the cosmos.


HUMAN-LIKE WRITING REQUIREMENTS:
- Plan for internal conflicts between main characters that won't resolve easily
- Identify opportunities for characters to fail meaningfully without plot advancement
- Consider morally ambiguous situations with no clear right answers
- Plan distinctive speech patterns and verbal tics for each major character
- Include mundane, lived-in world details that don't advance plot
- Design at least 2-3 unresolved elements to leave open
- Ensure antagonist has understandable (if wrong) motivations
- Plan for interpersonal friction based on fundamental value differences
- Create backstories with contradictory elements that surprise readers


UNIVERSAL FRAMEWORK FOR CHARACTER ANALYSIS:

CHARACTER FALLIBILITY REQUIREMENTS:
- Assign each main character a core psychological flaw that will drive poor decisions
- Identify how this flaw will cause meaningful failures with lasting consequences
- Plan scenarios where character strengths become weaknesses under pressure

ANTAGONIST COMPLEXITY:
- Develop antagonist as hero of their own story with understandable motivations
- Create opposition that makes valid points even when wrong overall
- Plan scenes showing antagonist's personal life, fears, and justifications

VALUE CONFLICT PLANNING:
- Design main characters with fundamentally opposing core values
- Plan situations where these values will clash and create lasting friction
- Identify choices that will force characters to act against their stated beliefs


LEVEL 5 ADVANCED FRAMEWORK FOR ANALYSIS:

FACTION BREAKING POINT PLANNING:
- Identify internal factions that can believably sabotage protagonist from good intentions
- Plan "hard choice" scenarios where leader must choose between control and collaboration
- Design consequences that last throughout story regardless of choice made
- Create conflicts where there are no clear villains, only competing needs

POWER COST INTEGRATION:
- Assign specific, painful costs to each character's special abilities or advantages
- Plan empathetic feedback loops where protectors suffer when their charges are harmed
- Design power limitations that create vulnerability and force creative problem-solving
- Ensure abilities never become convenient plot solutions without significant price

STRUCTURAL SUBVERSION OPPORTUNITIES:
- Identify established patterns in story structure that can be intentionally broken
- Plan critical decisions that must be made by isolated characters under pressure
- Design "milk run" scenarios that spiral into major consequences from small details
- Create unpredictability without sacrificing narrative coherence

Please provide a comprehensive analysis that prioritizes authentic, human-like storytelling:

ANALYSIS REQUIREMENTS:
1. Theme analysis - themes that allow for moral complexity
2. Character archetypes - with internal contradictions and growth potential
3. Plot structure - that accommodates meaningful failures and setbacks
4. Key story beats - including moments of genuine uncertainty
5. Potential subplots - that may remain partially unresolved
6. Tone and style guidance - that varies subtly throughout
7. HUMAN-LIKE ELEMENTS:
   - Internal character conflicts that create lasting tension
   - Opportunities for protagonist to be genuinely wrong
   - Morally ambiguous situations requiring difficult choices
   - Cultural/world elements that add lived-in authenticity
   - Distinctive character voice planning (speech patterns, vocabulary)

Respond in JSON format:
{
  "themes": ["theme1", "theme2"],
  "characters": [{"type": "character_type", "conflicts": "internal_struggles", "speechPattern": "distinctive_traits"}],
  "plotStructure": "structure_with_flexibility_for_messiness",
  "keyBeats": ["beat1", "beat2"],
  "subplots": [{"main": "subplot", "resolution": "complete|partial|unresolved"}],
  "tone": "description_that_allows_variation",
  "styleNotes": "guidance_for_character_specific_prose",
  "humanLikeElements": {
    "characterConflicts": ["lasting_disagreement1", "lasting_disagreement2"],
    "moralDilemmas": ["situation1", "situation2"],
    "culturalElements": ["in_world_reference1", "in_world_reference2"],
    "unresolvedElements": ["mystery1", "relationship_tension1"]
  }
}"
`;
